python/ray/dataframe/dataframe.py

```diff
@@ -2774,37 +2773,66 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
                     are the quantiles.
         """
 
-        def quantile_helper(df, q, axis, numeric_only, interpolation):
-            try:
-                return df.quantile(q=q, axis=axis, numeric_only=numeric_only,
-                                   interpolation=interpolation)
-            except ValueError:
-                return pd.Series()
+        # deal with error checking for bad dtyping when numeric_only is False
+        if not numeric_only:
```

Comment:
`numeric_only` specifies whether or not to filter the non-numeric columns. Does not need error checking.

Reply:
If numeric_only is true, then pandas returns value errors for mixed dtype dataframes

Reply:
I'm not able to reproduce, seems that pandas drops non-numeric columns when `numeric_only` is True.

```
In [3]: df
Out[3]: 
   col1 col2
0     1    a
1     2    b
2     3    c

In [4]: df.dtypes
Out[4]: 
col1     int64
col2    object
dtype: object

In [5]: df.quantile(numeric_only=True)
Out[5]: 
col1    2.0
Name: 0.5, dtype: float64
```

Reply:
This is true also in the `axis=1` case.
```
In [6]: df.quantile(numeric_only=True, axis=1)
Out[6]: 
0    1.0
1    2.0
2    3.0
Name: 0.5, dtype: float64
```

Reply:
Whoops, when numeric_only is FALSE, it returns TYPEerrors. 

Try on 

df = pd.DataFrame({"A": [1, 2, 3],
            "B": [2., 3., 4.],
            "C": pd.date_range('20130101', periods=3),
            "D": ['foo', 'bar', 'baz']})
df.quantile(.5, axis=1, numeric_only=False)

Reply:
This is true, although the check here does not handle this case. The error I see is `TypeError: Cannot compare type 'Timestamp' with type 'float'`.